src/main/java/org/jenkinsci/plugins/gitserver/HttpGitRepository.java

```diff
@@ -60,11 +63,27 @@ public HttpGitRepository() {
      *
      * @see ReceivePackFactory#create(Object, Repository)
      */
-    public abstract ReceivePack createReceivePack(HttpServletRequest context, Repository db) throws ServiceNotEnabledException, ServiceNotAuthorizedException;
+    @SuppressWarnings({"deprecated", "java:S1874"})
```

Comment:
Coming back to this PR, this is the only part I feel uneasy about. It implements the compatibility layer in the opposite way that Jenkins core did in jenkinsci/jenkins#9672, by having the new functionality delegate to the old rather than having the old functionality delegate to the new. This also makes it harder to eventually remove the old functionality. For consistency, I would want to see this compatibility layer implemented the same way as jenkinsci/jenkins#9672 or jenkinsci/cloudbees-folder#444.

Reply:
BTW, the closest analog I can find for this method (as I needed to change the signature of an abstract method, not a concrete method) is `ReconfigurableDescribable` in jenkinsci/jenkins#9672, as it's an interface and one of the interface methods needed to change. In that scenario, the interface method was changed into 2 default methods that both check if the other is present. I will take this approach, changing the abstract method into the 2 concrete methods I've already done, but by having both check for the presence of the other.

Reply:
`ParameterDefinition` is an example of an abstract method that doesn't throw an exception. For one that does, `View#submit` is an example.

Reply:
Thanks, I missed those ones. I think what I've gotten is quite close to `View#submit`.

Reply:
> having both check for the presence of the other

Probably unnecessary, since in this case no dependent plugin should be _calling_ this method, only _implementing_ it. So it should suffice to delegate in one direction only.